elastalert/test_rule.py

```diff
@@ -1,86 +1,285 @@
 from elastalert.elastalert import ElastAlerter
 from elastalert.elastalert import Elasticsearch
 from elastalert.util import lookup_es_key
+from elastalert.util import replace_hits_ts
 from elastalert.util import ts_now
+from elastalert.util import ts_to_dt
+
+logging.getLogger().setLevel(logging.INFO)
 
 
 def print_terms(terms, parent):
+    """ Prints a list of flattened dictionary keys """
+    logging.info("\nAvailable terms in first hit:")
     for term in terms:
         if type(terms[term]) != dict:
-            print('\t', parent + term)
+            logging.info('\t', parent + term)
         else:
             print_terms(terms[term], parent + term + '.')
 
 
-def check_files():
-    print("Note: This tool is for testing filters and config syntax. It will not process data or alert.\n")
-    parser = argparse.ArgumentParser(description='Validate a rule configuration')
-    parser.add_argument('files', metavar='file', type=str, nargs='+', help='rule configuration filename')
-    parser.add_argument('--schema-only', action='store_true', help='Show only schema errors; do not run query')
-    parser.add_argument('--days', type=int, default=[1, 7], nargs='+', help='Query the previous N days with this rule')
-    args = parser.parse_args()
+class MockElastAlerter(object):
+    def __init__(self):
+        self.data = []
 
-    for filename in args.files:
+    def test_file(self, args):
+        """ Loads a rule config file, performs a query over the last day (args.days), lists available keys
+        and prints the number of results. """
+        filename = args.file
         with open(filename) as fh:
             conf = yaml.load(fh)
         load_options(conf)
-        print("Successfully loaded %s\n" % (conf['name']))
+        logging.info("Successfully loaded %s\n" % (conf['name']))
 
         if args.schema_only:
-            continue
+            return []
 
+        # Set up elasticsearch client and query
         es_client = Elasticsearch(host=conf['es_host'], port=conf['es_port'])
-        for days in args.days:
-            start_time = ts_now() - datetime.timedelta(days=days)
-            end_time = ts_now()
-            ts = conf.get('timestamp_field', '@timestamp')
-            query = ElastAlerter.get_query(conf['filter'], starttime=start_time, endtime=end_time, timestamp_field=ts)
-            index = ElastAlerter.get_index(conf, start_time, end_time)
+        start_time = ts_now() - datetime.timedelta(days=args.days)
+        end_time = ts_now()
+        ts = conf.get('timestamp_field', '@timestamp')
+        query = ElastAlerter.get_query(conf['filter'], starttime=start_time, endtime=end_time, timestamp_field=ts)
+        index = ElastAlerter.get_index(conf, start_time, end_time)
+
+        # Get one document for schema
+        try:
+            res = es_client.search(index, size=1, body=query, ignore_unavailable=True)
+        except Exception as e:
+            logging.info("Error running your filter:")
```

Comment:
Not sure if .info is appropriate here (I'd prefer if we got stderr), but I can see why this is reasonable given how the tool is invoked.